qiime/sdk/tests/test_visualization.py

```diff
@@ -0,0 +1,244 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import zipfile
+import tempfile
+import unittest
+import uuid
+
+import qiime.core.type
+from qiime.sdk import Visualization, Provenance
+
+from qiime.core.testing.visualizer import mapping_viz
+from qiime.core.testing.util import get_data_path
+
+
+class TestVisualization(unittest.TestCase):
+    def setUp(self):
+        # TODO standardize temporary directories created by QIIME
```

Comment:
:+1: it might even be worth creating a qiime directory in the associated tempdir to keep things tidy.